auth/src/main/java/com/redhat/lightblue/rest/auth/ldap/LdapRolesProvider.java

```diff
@@ -190,4 +191,26 @@ private void initialize() throws Exception {
         return roles;
     }
 
+    @Override
+    public RolesProviderHealth checkHealth() {
+
+        boolean isHealthy = true;
+        String details = null;
+        
+        try {
+            LDAPConnection ldapConnection = connectionPool.getConnection();
+
+            if (!ldapConnection.isConnected()) {
+                isHealthy = false;
+            }
```

Comment:
Actually, since the default health check implementation may not do anything: I'd do both. First check `isConnected`, then call `ensureConnectionValidForContinuedUse()`. If both pass, we're healthy.